chain/neutrino.go

```diff
@@ -88,12 +93,19 @@ func (s *NeutrinoClient) Start() error {
 // Stop replicates the RPC client's Stop method.
 func (s *NeutrinoClient) Stop() {
 	s.clientMtx.Lock()
-	defer s.clientMtx.Unlock()
 	if !s.started {
+		s.clientMtx.Unlock()
 		return
 	}
-	close(s.quit)
 	s.started = false
+	s.clientMtx.Unlock()
+
+	if err := s.CS.Stop(); err != nil {
```

Comment:
Would it be better to stop it after all the goroutines have exited? This way we don't attempt to use it while it's shutting down.

Reply:
Hm, initially I wanted to avoid this in order to not change current behavior by waiting for the `wg` here. But maybe it is better to just do it, I agree.

Reply:
Done!

Reply:
So within `WaitForShutdown` then?

Reply:
Stop no moved out of `NeutrinoClient`, leaving the `WaitForShutdown` behavior intact.